### DISCUSSION TITLE
[question] Interpolating closed spline ?

### DISCUSSION THREAD

**Discussion by 4rzael** [ttk592/spline#13] [Q&A] [answered]
Hello,

Is this library able to interpolate a closed spline curve ?

Thanks

**Comment by ttk592** [comment, marked as answer]
The library lacks the ability to set the periodic boundary condition f'(a) = f'(b) which we would need for a closed curve which is smooth at the point where it connects. However, with a small workaround it can still be achieved as a reasonable approximation, see `examples/plot_parametric.cpp`. E.g. the below draws something resembling an inf sign:
```
$ ./plot_parametric.sh -x 0 0 1 2 2 1 -y 0 1 0.5 0 1 0.5 -c
```
![closed_curve](https://user-images.githubusercontent.com/1814760/108896170-d8950380-7614-11eb-9288-98ad9a5c4a2b.gif)


The correct way to do it:
* define a time grid points, e.g. `T[i+1]=T[i]+distance(X[i],Y[i], X[i+1],Y[i+1])`.
* set splines for x and y grid points: `sx.set_points(T,X);` and `sy.set_points(T,Y);` with periodic boundary conditions (which is not implemented)
* plot points `x=sx(t);` and `y=sy(t);` and let `t` go from `T[0]` to `T[n-1]`.

As a workaround to periodic boundary condition we can let the spline go through the closed curve a few times. It's a bit of a hack but it works well.
